src/sif/autocost.cc

```diff
@@ -498,6 +501,11 @@ Cost AutoCost::EdgeCost(const baldr::DirectedEdge* edge,
   auto edge_speed = tile->GetSpeed(edge, flow_mask_, time_info.second_of_week, false, &flow_sources,
                                    time_info.seconds_from_now);
   auto final_speed = std::min(edge_speed, top_speed_);
+
+  if (fixed_speed_ != 0) {
+    final_speed = fixed_speed_;
+  }
+
```

Comment:
in the case someone is using fixed speed its highly inefficient to still called `tile->GetSpeed()` just to throw the value away. maybe this would be better:

```suggestion
  auto edge_speed = fixed_speed_ != 0 ? tile->GetSpeed(edge, flow_mask_, time_info.second_of_week, false, 
                                   &flow_sources, time_info.seconds_from_now) : fixed_speed_;
  auto final_speed = std::min(edge_speed, top_speed_);
```

Reply:
Agreed much more efficient. Changed to save that calculation.